dapp/src/components/page/project/DonateModal.tsx

```diff
@@ -0,0 +1,203 @@
+import {
+  Children,
+  cloneElement,
+  useRef,
+  useState,
+  type FC,
+  type ReactElement,
+  type ReactNode,
+} from "react";
+
+import Modal from "components/utils/Modal";
+import Tooltip from "components/utils/Tooltip";
+import { toast } from "utils/utils";
+
+interface Props {
+  children: ReactNode;
+}
+
+const DonateModal: FC<Props> = ({ children }) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [amount, setAmount] = useState<number>(10);
+  const [tipAmount, setTipAmount] = useState<string>("");
+  const [donateMessage, setDonateMessage] = useState<string>("");
+  const amountInputRef = useRef<HTMLInputElement>(null);
+  const tipAmountInputRef = useRef<HTMLInputElement>(null);
+  const onClose = () => setIsOpen(false);
+
+  const amountOptions = [10, 100, 1000];
+
+  const handleAmountChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const value = Number(e.target.value);
+    setAmount(value);
+  };
+
+  const handleTipAmountChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setTipAmount(e.target.value.replace(/[^0-9]/g, ""));
+  };
+
+  const handleAmountButtonClick = (value: number) => {
+    setAmount(value);
+  };
+
+  const handleContribute = async () => {
+    const { sendXLM } = await import("service/PaymentService");
+    const { getAddressFromDomain } = await import(
+      "service/SorobanDomainContractService"
+    );
+
+    if (amount < 1) {
+      toast.error("Support", "Minimum donation is 1 XLM.");
+      return;
+    }
+
+    try {
+      const domainInfo = await getAddressFromDomain("tansu");
```

Comment:
No only the tip goes to Tansu. The amount must go to the current project. Like right now.